bitnami/redis/templates/master/application.yaml

```diff
@@ -445,11 +445,15 @@ spec:
             defaultMode: 0755
         {{- if .Values.auth.usePasswordFiles }}
         - name: redis-password
+          {{ if .Values.auth.usePasswordFileFromSecret }}
           secret:
             secretName: {{ template "redis.secretName" . }}
             items:
             - key: {{ template "redis.secretPasswordKey" . }}
               path: redis-password
+          {{- else }}
+          emptyDir: {}
+          {{- end }}
```

Comment:
A couple of questions:

- Should we add a warning in NOTES.txt if `auth.usePasswordFileFromSecret` is `false` and no custom init-containers are provided?
- What happens if both `auth.usePasswordFiles` & `auth.usePasswordFileFromSecret` are `false`? Do we expect the chart to provide some alternative mechanism to set `REDIS_PASSWORD` env. variable?

Reply:
> Should we add a warning in NOTES.txt if auth.usePasswordFileFromSecret is false and no custom init-containers are provided?

Updated in https://github.com/bitnami/charts/pull/23886/commits/539abf35a0f9ae291064e73f1d1bfc24b9c48bc8 .

> What happens if both auth.usePasswordFiles & auth.usePasswordFileFromSecret are false? Do we expect the chart to provide some alternative mechanism to set REDIS_PASSWORD env. variable?

Using the values.yaml file as is, this is my output (after fixing an issue in https://github.com/bitnami/charts/pull/23886/commits/3f9db3531b44d05b6c8eec3d9889a0c8a2c2e1f5):
```
# usePasswordFiles: true
# usePasswordFileFromSecret: true

[e] ~/charts/bitnami/redis (main) $ k get pods
NAME                       READY   STATUS    RESTARTS   AGE
bitnami-redis-master-0     1/1     Running   0          84s
bitnami-redis-replicas-0   1/1     Running   0          84s
bitnami-redis-replicas-1   1/1     Running   0          48s
bitnami-redis-replicas-2   1/1     Running   0          28s

[e] ~/charts/bitnami/redis (main) $ k get secret
NAME                                  TYPE                 DATA   AGE
bitnami-redis                         Opaque               1      88s
sh.helm.release.v1.bitnami-redis.v1   helm.sh/release.v1   1      88s
```

```
# usePasswordFiles: false
# usePasswordFileFromSecret: true

[e] ~/charts/bitnami/redis (main) $ k get pods
NAME                       READY   STATUS    RESTARTS      AGE
bitnami-redis-master-0     1/1     Running   0             111s
bitnami-redis-replicas-0   1/1     Running   1 (69s ago)   110s
bitnami-redis-replicas-1   1/1     Running   0             49s
bitnami-redis-replicas-2   1/1     Running   0             29s

[e] ~/charts/bitnami/redis (main) $ k get secrets
NAME                                  TYPE                 DATA   AGE
bitnami-redis                         Opaque               1      114s
sh.helm.release.v1.bitnami-redis.v1   helm.sh/release.v1   1      114s
```

```
# usePasswordFiles: true
# usePasswordFileFromSecret: false
# Init container: not specified

[e] ~/charts/bitnami/redis (main) $ k get pods
NAME                       READY   STATUS    RESTARTS      AGE
bitnami-redis-master-0     1/1     Running   0             2m33s
bitnami-redis-replicas-0   0/1     Running   3 (17s ago)   2m33s
redis-client               1/1     Running   0             3d21h

[e] ~/charts/bitnami/redis (main) $ k get secrets
NAME                                  TYPE                 DATA   AGE
sh.helm.release.v1.bitnami-redis.v1   helm.sh/release.v1   1      2m37s

[e] ~/charts/bitnami/redis (main) $ k logs -f bitnami-redis-replicas-0
...
1:S 26 Feb 2024 11:22:40.282 # Error reply to PING from master: '-DENIED Redis is running in protected mode because protected mode is enabled and no password is set for the default user. In this mode connections are only accepted from the loopback interface. (...)'
...
```

```
# usePasswordFiles: true
# usePasswordFileFromSecret: false
# Init container: defined with the following content (simulating that it comes from another source --> ie. vault)

  initContainers:
    - name: toolsidecar
      image: busybox
      imagePullPolicy: IfNotPresent
      resources:
        limits:
          cpu: 100m
          memory: 400Mi
        requests:
          cpu: 50m
          memory: 100Mi
      volumeMounts:
        - name: redis-password
          mountPath: /opt/bitnami/redis/secrets
      command: ['sh', '-c', 'echo "testpassword" > /opt/bitnami/redis/secrets/redis-password']

[e] ~/charts/bitnami/redis (main) $ k get pods
NAME                       READY   STATUS    RESTARTS   AGE
bitnami-redis-master-0     1/1     Running   0          109s
bitnami-redis-replicas-0   1/1     Running   0          99s
bitnami-redis-replicas-1   1/1     Running   0          63s
bitnami-redis-replicas-2   1/1     Running   0          38s

[e] ~/charts/bitnami/redis (main) $ k get secrets
NAME                                  TYPE                 DATA   AGE
sh.helm.release.v1.bitnami-redis.v1   helm.sh/release.v1   1      110s

[e] ~/charts/bitnami/redis (main) $ k exec -ti bitnami-redis-master-0 -c redis -- cat /opt/bitnami/redis/secrets/redis-password
testpassword
```

```
# usePasswordFiles: false
# usePasswordFileFromSecret: false

[e] ~/charts/bitnami/redis (main) $ k get pods
NAME                       READY   STATUS    RESTARTS   AGE
bitnami-redis-master-0     1/1     Running   0          117s
bitnami-redis-replicas-0   1/1     Running   0          117s
bitnami-redis-replicas-1   1/1     Running   0          86s
bitnami-redis-replicas-2   1/1     Running   0          61s

[e] ~/charts/bitnami/redis (main) $ k get secrets
NAME                                  TYPE                 DATA   AGE
bitnami-redis                         Opaque               1      119s
sh.helm.release.v1.bitnami-redis.v1   helm.sh/release.v1   1      119s
```